src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1569,6 +1569,36 @@ func handleDmVerityBlockVolume(driverType, source string, verityInfo *types.DmVe
 	return vol, nil
 }
 
+func handleImageGuestPullBlockVolume(c *Container, virtualVolumeInfo *types.KataVirtualVolume, vol *grpc.Storage) (*grpc.Storage, error) {
+	container_annotations := c.GetAnnotations()
+	container_type := container_annotations["io.kubernetes.cri.container-type"]
+	if virtualVolumeInfo.Source != "" {
```

Comment:
How about simplifying it as `if virtualVolumeInfo.Source == ""`?

Reply:
Fixed.